airbyte-cdk/python/airbyte_cdk/connector_builder/migrator/source.py

```diff
@@ -0,0 +1,166 @@
+
+    def _fetch_spec_outside_manifest(self, source_name: str) -> Optional[Spec]:
+        json_spec_path = os.path.join(self._python_package_path(source_name), "spec.json")
+        yaml_spec_path = os.path.join(self._python_package_path(source_name), "spec.yaml")
+
+        if os.path.exists(json_spec_path):
+            with open(json_spec_path) as json_file:
+                return self._assemble_spec(json.load(json_file))
+        elif os.path.exists(yaml_spec_path):
+            with open(yaml_spec_path) as yaml_file:
+                return self._assemble_spec(yaml.safe_load(yaml_file))
+        else:
+            return None
+
+    @staticmethod
+    def _assemble_spec(spec: Dict[str, Any]) -> Spec:
+        return Spec(
+            type="Spec",
+            connection_specification=spec["connectionSpecification"],
+            documentation_url=spec.get("documentationUrl", None),
+            advanced_auth=AuthFlow(**spec["advanced_auth"]) if "advanced_auth" in spec else None
+        )
+
+    def _delete_spec_outside_manifest(self, source_name: str) -> None:
+        json_spec_path = os.path.join(self._python_package_path(source_name), "spec.json")
+        yaml_spec_path = os.path.join(self._python_package_path(source_name), "spec.yaml")
+
+        if os.path.exists(json_spec_path):
+            os.remove(json_spec_path)
+        elif os.path.exists(yaml_spec_path):
+            os.remove(yaml_spec_path)
+
+    def fetch_no_code_sources(self) -> List[str]:
+        no_code_sources = []
+
+        connectors_folder_path = os.path.join("airbyte-integrations", "connectors")
+        for directory in os.listdir(connectors_folder_path):
+            manifest_file_path = os.path.join(connectors_folder_path, directory, directory.replace("-", "_"), "manifest.yaml")
+            if self._is_no_code_connector(manifest_file_path):
+                no_code_sources.append(directory)
+        return no_code_sources
+
+    @staticmethod
+    def _is_no_code_connector(manifest_file_path: str) -> bool:
+        if not os.path.exists(manifest_file_path):
+            return False
+
+        with open(manifest_file_path) as manifest_file:
+            file_content = manifest_file.read()
+            # We can't rely on the `type: Custom*` as some components do not require this tag. We therefore assume that if there is a field
+            # `class_name`, it'll be a custom component
+            return not re.search("class_name:", file_content)
+
+    def _manifest_path(self, source_name: str) -> str:
+        return os.path.join(self._python_package_path(source_name), "manifest.yaml")
+
+    def _python_package_path(self, source_name: str) -> str:
+        return os.path.join(self._path_from_name(source_name), source_name.replace("-", "_"))
+
+    def _path_from_name(self, source_name: str) -> str:
+        return os.path.join(self._base_path, "airbyte-integrations", "connectors", source_name)
+
+
+class _IndentingEmitter(yaml.emitter.Emitter):
```

Comment:
Using this indentation as it is what the UI uses and will therefore reduce the variability for users